os/arch/arm/src/armv8-m/up_releasepending.c

```diff
@@ -118,6 +122,18 @@ void up_release_pending(void)
 			 */
 
 			up_savestate(rtcb->xcp.regs);
+#ifdef CONFIG_ARMV8M_TRUSTZONE
```

Comment:
There are repeated two functions.
```
#ifdef CONFIG_ARMV8M_TRUSTZONE
			/* If there is associated secure context
			 * with the current active task, then load the secure context
			 */
			tz_memory = rtcb->xcp.regs[REG_R8];
			uint32_t result = rtcb->xcp.regs[REG_R9];
			set_PSPLIM(result);
			if (tz_memory) {
				TZ_LoadContext_S(tz_memory);
			}
#endif
```
and
```
#ifdef CONFIG_ARMV8M_TRUSTZONE
			/* Yes, then we should check if there is associated secure context
			 * with the current active task. If yes, then store the secure context
			 * before saving the non-secure context
			 */
			if (tz_memory) {
				TZ_StoreContext_S(tz_memory);
			}
			rtcb->xcp.regs[REG_R8] = tz_memory;
			rtcb->xcp.regs[REG_R9] = get_PSPLIM();
			tz_memory = 0;
#endif
```

How about making them as functions or definitions to remove duplication?
Code duplication is checking by SAM (Software Architecture Maturity).

Reply:
ok.